src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs

```diff
@@ -108,11 +108,33 @@ public async Task ContainerStepAsync(IExecutionContext context)
             throw new NotImplementedException();
         }
 
-        public async Task RunScriptStepAsync(IExecutionContext context)
+        public async Task RunScriptStepOnJobContainerAsync(IExecutionContext context, ContainerInfo container, string entryPointArgs, string entryPoint, IDictionary<string, string> environmentVariables, string prependPath, string workingDirectory)
         {
             Trace.Entering();
-            await Task.FromResult(0);
-            throw new NotImplementedException();
+            var responsePath = GenerateResponsePath();
+
+            context.JobContext.TryGetValue("hook_state", out var hookState);
+            var input = new HookInput
+            {
+                Command = HookCommand.RunScriptStep,
+                ResponseFile = responsePath,
+                Args = new HookStepArgs
+                {
+                    Container = container.GetHookContainer(),
+                    EntryPointArgs = entryPointArgs.Split(' ').Select(arg => arg.Trim()),
+                    EntryPoint = entryPoint,
+                    EnvironmentVariables = environmentVariables,
+                    PrependPath = prependPath,
+                    WorkingDirectory = workingDirectory,
+                },
+                State = JsonUtility.FromString<dynamic>(hookState.ToString())
```

Comment:
Also nullcheck (maybe a method `GetHookState`?)

Reply:
I'll make a method `GetHookState` and just call it.